src/os/bluestore/BlueStore.h

```diff
@@ -569,10 +573,13 @@ class BlueStore : public ObjectStore,
 
   /// a cache (shard) of onodes and buffers
   struct Cache {
+    BlueStore *store;
     std::mutex lock;                ///< protect lru and other structures
 
     static Cache *create(string type);
 
+    void set_store(BlueStore *bs) { store = bs; }
```

Comment:
IMHO this 
+    logger->inc(l_bluestore_onode_misses);
just needs to be replaced with 
+    cache->logger->inc(l_bluestore_onode_misses);
The same at the second location..